### ISSUE TITLE
Query for specific affiliates

### ISSUE THREAD

**Issue by DrewAPicture** [AffiliateWP/AffiliateWP#1220]
Just as we have the ability to query for specific users corresponding to affiliates, we should also be able to query for specific affiliates.

**Comment by michaelbeil** [comment]
As in specific affiliates corresponding to users?

**Comment by DrewAPicture** [comment]
No, as in specific affiliates. We should be able to query for specific affiliate objects by ID. We currently have logic to pull a single affiliate by ID but not more than one short of doing a foreach for a `->get_by()` call.

**Comment by DrewAPicture** [comment]
To pull by a single user or multiple users:
```
$results = affiliate_wp()->affiliates->get_affiliates( array(
	'user_id' => 1
) );

$results = affiliate_wp()->affiliates->get_affiliates( array(
	'user_id' => array( 1, 2 )
) );
```

But there's currently no way to query for single or multiple affiliates by ID with `get_affiliates()`, e.g.
```
$results = affiliate_wp()->affiliates->get_affiliates( array(
	'affiliate_id' => 1
) );

$results = affiliate_wp()->affiliates->get_affiliates( array(
	'affiliate_id' => array( 1, 2 )
) );
```

**Comment by michaelbeil** [comment]
Oh, I see now. Seems like a great addition.

**Comment by DrewAPicture** [comment]
You could do something like this in a pinch:
```
$affiliates = array();
foreach ( array( 1, 2 ) as $affiliate_id ) {
	$affiliates[] = affiliate_wp()->affiliates->get_by( 'affiliate_id', $affiliate_id );
}
```
But I think this should be supported out of the box :)

**Comment by DrewAPicture** [comment]
Added a PR: #1221

**Comment by DrewAPicture** [comment]
Merged #1221.
